pkg/apis/core/validation/validation_test.go

```diff
@@ -5187,20 +5187,20 @@ func TestValidatePod(t *testing.T) {
 			},
 			Spec: validPodSpec(nil),
 		},
-		{ // valid opaque integer resources for init container
-			ObjectMeta: metav1.ObjectMeta{Name: "valid-opaque-int", Namespace: "ns"},
+		{ // valid extended resources for init container
+			ObjectMeta: metav1.ObjectMeta{Name: "valid-extended", Namespace: "ns"},
 			Spec: core.PodSpec{
 				InitContainers: []core.Container{
 					{
-						Name:            "valid-opaque-int",
+						Name:            "valid-extended",
 						Image:           "image",
 						ImagePullPolicy: "IfNotPresent",
 						Resources: core.ResourceRequirements{
 							Requests: core.ResourceList{
-								helper.OpaqueIntResourceName("A"): resource.MustParse("10"),
+								core.ResourceName("example.com/a"): resource.MustParse("10"),
 							},
 							Limits: core.ResourceList{
-								helper.OpaqueIntResourceName("A"): resource.MustParse("20"),
+								core.ResourceName("example.com/a"): resource.MustParse("10"),
```

Comment:
why change 20 to 10？

Reply:
This test case is to verify the handling of valid extended resource requests. One difference between extended resources and opaque integer resources is that extended resources may not be overcommitted. So, if both request and limit are specified then they must be equal to pass validation.

Reply:
We have other test cases in this file where extended resource request != limit, but we expect them to fail validation.